### PR TITLE
Add request timing instrumentation hooks

### PR DESCRIPTION
**Pull request by skeggse**

### CODE REVIEW

File: lib/http-api.js [hunk 1]
```diff
@@ -77,11 +77,16 @@ HttpApi.prototype.request = function(request, callback) {
 
   self.emit('request', request);
   logger.debug("<request> method=" + request.method + ", url=" + request.url);
-  var requestTime = Date.now();
+  const requestTime = process.hrtime();
+  const onComplete = () => {
+    const elapsedTime = process.hrtime(requestTime);
+    const duration = elapsedTime[0] * 1e3 + ((elapsedTime[1] / 1e6) | 0);
```
Comment (mericsson) [inline lib/http-api.js]: can you add comment here about why we are doing this?

**Comment by mericsson** [comment]
added comment, otherwise lgtm!

File: lib/http-api.js [hunk 2]
```diff
@@ -97,8 +104,7 @@ HttpApi.prototype.request = function(request, callback) {
     }
     return self.getResponseBody(response);
   }, function(err) {
-    var responseTime = Date.now();
-    logger.debug("elappsed time : " + (responseTime - requestTime) + "msec");
+    onComplete();
```
Comment (shils) [inline lib/http-api.js]: would upstream listeners be able to distinguish error response times from success response times?

Comment (skeggse) [reply lib/http-api.js]: not currently; we're just looking for overall timing

**Pull request merged**